shared/Microsoft.Extensions.WebEncoders.Sources/WebEncoders.cs

```diff
@@ -56,23 +65,102 @@ public static byte[] Base64UrlDecode(string input)
+        /// <remarks>
+        /// The input must not contain any whitespace or padding characters.
+        /// Throws <see cref="FormatException"/> if the input is malformed.
+        /// </remarks>
+        public static int Base64UrlDecode(ReadOnlySpan<char> base64Url, Span<byte> data)
+        {
+            // Special-case empty input
+            if (base64Url.IsEmpty)
+            {
+                return 0;
+            }
+
+            var base64Len = GetBufferSizeRequiredToUrlDecode(base64Url.Length, out int dataLength);
+            var written = Base64UrlDecodeCore(base64Url, data, base64Len);
+            Debug.Assert(dataLength == written);
+
+            return written;
+        }
+
+        /// <summary>
+        /// Decode the span of UTF-8 base64url-encoded text into binary data.
+        /// </summary>
+        /// <param name="base64Url">The input span which contains UTF-8 base64url-encoded text that needs to be decoded.</param>
+        /// <param name="data">The output span which contains the result of the operation, i.e. the decoded binary data.</param>
+        /// <param name="bytesConsumed">The number of input bytes consumed during the operation. This can be used to slice the input for subsequent calls, if necessary.</param>
+        /// <param name="bytesWritten">The number of bytes written into the output span. This can be used to slice the output for subsequent calls, if necessary.</param>
+        /// <param name="isFinalBlock">True (default) when the input span contains the entire data to decode.
+        /// Set to false only if it is known that the input span contains partial data with more data to follow.</param>
+        /// <returns>It returns the OperationStatus enum values:
+        /// - Done - on successful processing of the entire input span
+        /// - DestinationTooSmall - if there is not enough space in the output span to fit the decoded input
+        /// - NeedMoreData - only if isFinalBlock is false and the input is not a multiple of 4, otherwise the partial input would be considered as InvalidData
+        /// - InvalidData - if the input contains bytes outside of the expected base 64 range, or if it contains invalid/more than two padding characters,
+        ///   or if the input is incomplete (i.e. not a multiple of 4) and isFinalBlock is true.</returns>
+        public static OperationStatus Base64UrlDecode(ReadOnlySpan<byte> base64Url, Span<byte> data, out int bytesConsumed, out int bytesWritten, bool isFinalBlock = true)
+        {
+            // Special-case empty input
+            if (base64Url.IsEmpty)
+            {
+                bytesConsumed = 0;
+                bytesWritten = 0;
+                return OperationStatus.Done;
+            }
+
+            var base64Len = isFinalBlock
+                ? GetBufferSizeRequiredToUrlDecode(base64Url.Length, out int dataLength)
+                : base64Url.Length;
+
+            if (base64Len > MaxStackallocBytes / sizeof(byte))
```

Comment:
Isn't sizeof(byte) 1? Why do we need this division?

Reply:
It is redundant, and just to have it the same as on the char-part. The JIT will treat it as constant anyway.